Sources/Swift/Core/Helper/SentryExtraPackages.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+@objc @_spi(Private) public final class SentryExtraPackages: NSObject {
+    private static var extraPackages = Set<[String: String]>()
+    private static let syncQueue = DispatchQueue(label: "io.sentry.SentryExtraPackages.sync", attributes: .concurrent)
+
+    @objc
+    public static func addPackageName(_ name: String?, version: String?) {
+        guard let name, let version else {
+            return
+        }
+
+        let newPackage = ["name": name, "version": version]
+
```

Comment:
`l`: Add a comment explaining why `async(flags: .barrier)` is used here.

```suggestion

        // Thread-safe write using a barrier block on a concurrent queue.
        // Ensures no other reads or writes happen during this insertion.
```